FWCore/Services/bin/edmTracerCompactLogViewer.py

```diff
@@ -0,0 +1,577 @@
+#!/usr/bin/env python3
+from __future__ import print_function
+from builtins import range
+from itertools import groupby
+from operator import attrgetter,itemgetter
+import sys
+from collections import defaultdict
+#----------------------------------------------
+def printHelp():
+    s = '''
+To Use: Add the Tracer Service to the cmsRun job use something like this
+  in the configuration:
+
+  process.add_(cms.Service("Tracer", fileName = cms.untracked.string("tracer.log")))
+
+  After running the job, execute this script and pass the name of the
+  Tracer log file to the script.
+
+  By default, the script will then print an 'ASCII art' stall graph
+  which consists of a line of text for each time a module or the
+  source stops or starts. Each line contains the name of the module
+  which either started or stopped running, and the number of modules
+  running on each stream at that moment in time. After that will be
+  the time and stream number. Then if a module just started, you
+  will also see the amount of time the module spent between finishing
+  its prefetching and starting.  The state of a module is represented
+  by a symbol:
+
+    plus  ("+") the stream has just finished waiting and is starting a module
+    minus ("-") the stream just finished running a module
+
+  If a module had to wait more than 0.1 seconds, the end of the line
+  will have "STALLED". Startup actions, e.g. reading conditions,
+  may affect results for the first few events.
+
+  Using the command line arguments described above you can make the
+  program create a PDF file with actual graphs instead of the 'ASCII art'
+  output.
+
+  Once the graph is completed, the program outputs the list of modules
+  which had the greatest total stall times. The list is sorted by
+  total stall time and written in descending order. In addition, the
+  list of all stall times for the module is given.
+
+  There is an inferior alternative (an old obsolete way).
+  Instead of using the StallMonitor Service, you can use the
+  Tracer Service.  Make sure to use the 'printTimestamps' option
+  cms.Service("Tracer", printTimestamps = cms.untracked.bool(True))
+  There are problems associated with this and it is not recommended.'''
```

Comment:
How accurate is this help message? By quick look it looks mostly being copied from `edmStreamStallGrapher.py`.

Reply:
fixed